### PR TITLE
Fix hub timers

### LINKED ISSUE #71331: hub01 will provide prototype armor at no cost.
### Describe the bug

When purchasing armor in a dialogue with hub01, if you select a type and cancel, you will be offered the selected type of armor free of charge on the spot.

### Attach save file

N/A

### Steps to reproduce

1. Interact with the intercom on hub01 and select "Let's discuss prototypes."
2. 「I'd like to request a set of ...」
Select one of the
「I'll check in later.」
to end the conversation.
3. Interact with the Intercom again,
「Let's discuss prototypes.」
Select Create.
4. "How is my ordered armor going?  It should be done by now."
By selecting this option, you can receive the armor you selected in 2 for free.

### Expected behavior

hub01 should receive the appropriate rewards and should be required to take the time to fabricate the armor.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.22621.3085 (22H2)
- Game Version: b3a2e0d [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

### Additional context

I found some fixes for this issue.
In robofac_intercom_prototypes.json,
Line 24 { "math": [ "hub01_uhmwpe_researched", "==", "1" ] }
{ "math": [ "hub01_uhmwpe_researched", "!=", "1" ] },
Line 187 { "math": [ "time_since(u_timer_hub_rnd_u_waiting_on_hub)", "<", "time('1 d')" ] }
{ "math": [ "time_since(u_timer_hub_rnd_u_waiting_on_hub)", ">", "time('1 d')" ] }
Lines 576, 598, 620, 642, 664
{ "math": [ "u_timer_hub_rnd_u_waiting_on_gun", "=", "time('now')" ] }
Change to { "math": [ "u_timer_hub_rnd_u_waiting_on_hub", "=", "time('now')" ] }.
With this change, the option to order prototype armor for the first time worked properly.
However, from the second time onwards, the armor will be provided for free by following steps 2, 3, and 4 of Steps to reproduce.
This is expected to be due to the fact that the timer continues to count even after the first order is completed, and the variable saved with the option selected in step 2. However, I couldn't successfully fix this.

Finally, I relied on automatic translation to create this sentence.
I apologize if there are any mistakes in the text.

### LINKED ISSUE #71366: Hub prototype system broken
### Describe the bug

the prototype system with the hub seems to be broken. For the gun it seems to work sometimes but for all my testing the UHMWPE armor is unobtainable. You finish the related quest for the armor hand in the extra disc and get told to wait after 24 hours they list the disk for sale and you cant ask about it and you dont get given anything. Finishing the gun quest seems to have a 50/50 chance of actually starting the prototype timer and not soft locking you from hub progression..

### Attach save file

[Test World-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/14103939/Test.World-trimmed.tar.gz)

### Steps to reproduce

1. complete light retrieval
2. hand in UHMWPE disc
3. wait 24 hours
4. attempt to ask hum about prototype

### Expected behavior

After finishing light retrieval, handing in the UHMWPE disc and waiting for the hub to decode it to gain access to the upgraded hub armor

### Screenshots

_No response_

### Versions and configuration

OG: Win 11 Pro
Game Version: 1949d26
Graphics Version: Tile
Language: English
Mods Loaded: 	Disable NPC Needs [no_npc_food]
		Portal Storms Ignore NPCs [personal_portal_storms]
		Slowdown Fungal Growth [no_fungal_growth]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
Attempt to fix #71331 and fix #71366
#### Describe the solution
Swapped checks to return true when it passed **more** than X days, not less
replace waiting_on_gun with waiting_on_hub variable, because it looks like the second should respond for your armor crafting
#### Testing
I opened prototypes menu before the fix, confirmed i can finish it instantly
I applied fix, i tried again, and i can't do this anymore
#### Additional context
I don't really know is it suffisient fix, #71331 meniton error in Line 24, but i don't see how it may cause issues, and i am not entirely sure i undersand the logic behind all the dialogues

### CODE REVIEW

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json [hunk 1]
```diff
@@ -261,7 +261,7 @@
         "text": "How are my armor repairs coming?",
         "condition": {
           "and": [
-            { "math": [ "time_since(u_timer_hub_rnd_u_waiting_on_hub)", "<", "time('1 d')" ] },
+            { "math": [ "time_since(u_timer_hub_rnd_u_waiting_on_hub)", ">", "time('1 d')" ] },
```
Comment (andrei8l) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json:264]: This is for "it's not ready yet" so the sign is correct

**Comment by andrei8l** [comment]
I didn't notice  you had already put up a PR. There are a couple of other errors there that I've addressed in #71374

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json (hunk 1 above)
Comment (GuardianDll) [reply data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json:264]: Does the second commit correct? I assume the idea is to have separate timers for weapon and for armor, but didn't dig deep enough

**Comment by GuardianDll** [comment]
Nevertheless, it'd be easier to close it in favor of #71374

**Pull request closed without merging**